sat/apiclient/jobstat.py

```diff
@@ -0,0 +1,62 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Client for querying the Job State Checker service
+"""
+
+import logging
+
+from sat.apiclient.gateway import APIError, APIGatewayClient
+
+LOGGER = logging.getLogger(__name__)
+
+
+class JobstatClient(APIGatewayClient):
+    base_resource_path = 'statechecker/jobstat/'
+
+    def get_all(self):
+        """Get all results from Job State Checker.
+
+        Returns:
+            A list of dictionaries where each dictionary pertains to a
+            single job.
+
+        Raises:
+            APIError: if there is a failure querying the Job State Checker API
+                or getting the required information from the response.
+        """
+        err_prefix = 'Failed to get State Checker data'
+        try:
+            response = self.get('all').json()
+            if 'errorcode' in response and response['errorcode'] != 0:
+                raise APIError(response['errormessage'])
+            if 'jobstat' in response:
+                return response['jobstat']
+            return None
```

Comment:
This line is not necessary because functions will return `None` unless otherwise specified. If we want a default case, I'd make it an empty list, but I would prefer if the `jobstat` key is missing to be raising an error (as I mentioned in the comment above).

The reason I am thinking of the `errorcode` and `jobstat` fields as different in terms of how we should be checking them is that in the case that the `errorcode` field is missing but `jobstat` key is not, I think we should be able to return the jobstat data without an error. But in the cases:
* when both the `jobstat` key and the `errorcode` key are missing
* when the `errorcode` was 0 but the `jobstat` key was still missing

These are cases when I think we should be raising an error.

Reply:
I don't see this happening with the current server implementation. Only memory corruption could result in such scenarios.

Reply:
My worry is that the server implementation could change. I don't think there's ever really a good case for returning `None` from this function, I think it should either return valid results or raise an error.